.github/workflows/codecov.yml

```diff
@@ -0,0 +1,38 @@
+# Based on https://github.com/codecov/codecov-action?tab=readme-ov-file#example-workflowyml-with-codecov-action
+name: Codecov
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+jobs:
+  run:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+    env:
+      OS: ${{ matrix.os }}
+      PYTHON: '3.10'
+    steps:
+    - uses: actions/checkout@main
+    - name: Setup Python
+      uses: actions/setup-python@main
+      with:
+        python-version: '3.10'
+    - name: Generate coverage report
+      run: |
+        pip install pytest
+        pip install pytest-cov
+        pytest --cov=./ --cov-report=xml
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v4
+      with:
+        directory: ./coverage/reports/
+        env_vars: OS,PYTHON
+        fail_ci_if_error: true
+        files: ./coverage1.xml,./coverage2.xml,!./cache
```

Comment:
I don't think we need special named files, or do we? I though the `--cov-report=xml` genrated in default file `coverage.xml`

Reply:
Screenshot from your workflow

![image](https://github.com/user-attachments/assets/d0174f1f-b06f-4659-afe6-412529b8e0ca)

Reply:
I will remove this line about files